File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        Map<String, Integer> pocketmonToNumList = new HashMap<>();
        Map<Integer, String> numToPocketnumList = new HashMap<>();

        for (int i = 0; i < n; i++) {
            String name = br.readLine();
            pocketmonToNumList.put(name, (i + 1));
            numToPocketnumList.put(i + 1, name);
        }

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < m; i++) {
            String question = br.readLine();
//            번호로 물어봣다면
            if (Character.isDigit(question.charAt(0))) {
                int num = Integer.parseInt(question);
                sb.append(numToPocketnumList.get(num)).append("\n");
            } else {
                sb.append(pocketmonToNumList.get(question)).append("\n");
            }
        }

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val st = StringTokenizer(br.readLine())

    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()

    val numToPocketName = mutableMapOf<Int, String>()
    val nameToPocketNum = mutableMapOf<String, Int>()

    for (i in 0 until n) {
        val pocketname = br.readLine()
        numToPocketName.put((i + 1), pocketname)
        nameToPocketNum.put(pocketname, (i + 1))
    }

    val sb = StringBuilder()
    for (i in 0 until m) {
        val question = br.readLine()
        if (Character.isDigit(question[0])) {
            val num = question.toInt()
            sb.append(numToPocketName.get(num))
        } else {
            sb.append(nameToPocketNum.get(question))
        }
        sb.append("\n")
    }

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}